Make it work on Ubuntu 16.04

qznc opened the pull request:
Requires two fixes:

* mktemp requires "XXX"
* Ubuntu git is at version 2.7.4, which does not yet have `--allow-unrelated-histories`

Commit:
Fix handling of --allow-unrelated-histories

Previous to git 2.9.0 the parameter would yield an error.
From git 2.9.0 this parameter is required.

Commit:
Fix mktemp requiring "XXX" for random name generation

Fixes #1

jeremysears commented on bin/git-submodule-rewrite:
```diff
@@ -50,6 +50,11 @@ EOF
 
+function git_version_lte() {
+  GIT_VERSION=$(git version)
+  [ "$1" = $(echo -e "${GIT_VERSION#git version}\n$1" | sort -V | head -n1) ]
```
I should probably have mentioned this somewhere, but this was only tested on MacOS.  Unfortunately, sort on MacOS doesn't support a '-V' option for semver sorting.  It's cool to know it exists elsewhere, though, so I'll be using that with gnusort.  I'm not sure what options are available otherwise.  There are a few hacks available here:
http://stackoverflow.com/questions/21394536/how-to-simulate-sort-v-on-mac-osx

Thoughts?

jeremysears reviewed (changes requested):
Thanks for contributing this, and I'm glad to hear you found this helpful.  What do you think about the issue I mentioned above.

jeremysears reviewed (approved):
Looks great.  I think this just yields the following on MacOS, but I think that's acceptable:
/<prefix>/submodule-rewrite-XXXXXX.Rq3yADjT

Commit:
Fix for OS X hopefully

OS X sort does not have the "-V" argument for semver ordering. Used
another comparison from http://stackoverflow.com/a/29394504/2361979

qznc commented:
Tried [another comparison method](http://stackoverflow.com/a/29394504/2361979). Does that work on OS X?

jeremysears commented:
That worked.  I've merged your changes.  Thanks for that!
